project/Common.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka
+
+import sbt._
+import Keys._
+
+object Common extends AutoPlugin {
+  override def trigger: PluginTrigger = allRequirements
+  override lazy val projectSettings = Seq(
+    scalacOptions ++= Seq(
+      "-deprecation",
+      "-encoding", "UTF-8", // yes, this is 2 args
+      "-target:jvm-1.8",
+      "-unchecked",
+      "-Xlint",
+      // "-Yno-adapted-args", //akka-http heavily depends on adapted args and => Unit implicits break otherwise
+      "-Ywarn-dead-code"
+      // "-Xfuture" // breaks => Unit implicits
+    ),
+    // '-release' parameter is restricted to 'Compile, compile' scope because
+    // otherwise `sbt akka-http-xml/compile:doc` fails with it on Scala 2.12.9
+    scalacOptions in (Compile, compile) ++= {
+      if (scalaMinorVersion.value >= 12 && !sys.props("java.version").startsWith("1.") /* i.e. Java version >= 9 */)
+        Seq("-release", "8")
+      else
+        Nil
+    },
+    javacOptions ++= Seq(
+      "-encoding", "UTF-8",
+    ),
+    javacOptions ++= {
+      if (!sys.props("java.version").startsWith("1.") /* i.e. Java version >= 9 */)
```

Comment:
Maybe extract that method now that it is used so much (also it's somewhat subtle).